### PR TITLE
[5/n][pipeline-gen] Method to read test steps from Yaml file

### PR DESCRIPTION
**Pull request by khluu**

### CODE REVIEW

File: scripts/pipeline_generator/pipeline_generator.py [hunk 1]
```diff
@@ -53,3 +52,24 @@ def __init__(
...
+
+def read_test_steps(file_path: str) -> List[TestStep]:
+    """Read test steps from test pipeline yaml and parse them into TestStep objects."""
+    with open(file_path, "r") as f:
+        content = yaml.safe_load(f)
+    return [TestStep(**step) for step in content["steps"]]
+
+@click.command()
+@click.option("--test_path", type=str, required=True, help="Path to the test pipeline yaml file")
+@click.option("--run_all", type=str)
```
Comment (aslonnie) [inline scripts/pipeline_generator/pipeline_generator.py:64]: maybe add help string that says `"1"` means true? and all other values means false?

File: scripts/pipeline_generator/pipeline_generator.py [hunk 2]
```diff
@@ -53,3 +52,24 @@ def __init__(
...
+@click.option("--list_file_diff", type=str)
```
Comment (aslonnie) [inline scripts/pipeline_generator/pipeline_generator.py:65]: is this a file or a diff? could you add a help for this too?

**Review by aslonnie — APPROVED** [review]

File: scripts/pipeline_generator/pipeline_generator.py (hunk 1 above)
Comment (khluu) [reply scripts/pipeline_generator/pipeline_generator.py]: done

File: scripts/pipeline_generator/pipeline_generator.py (hunk 2 above)
Comment (khluu) [reply scripts/pipeline_generator/pipeline_generator.py]: done

**Pull request merged**